backend/controllers/passController.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler")
+require("dotenv").config()
+const { connectDB } = require("../config/db")
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const { name, email, phone, duration, start } = req.body
+
+    const user = await User.findOne({ _id: req.user._id })
+    if (user) {
+      const newPass = await Pass.create({
+        name,
+        phone,
+        email,
+        duration,
+        start,
+        checkInStatus: false,
+        generatedUserId: user.id,
+        userName: req.user.name,
+      })
+      if (newPass) {
+        res.status(201).json({
+          success: true,
+          newPass: newPass,
+          message: "Pass generated successfully",
+        })
+      }
+    } else {
+      throw new ErrorHandler(401, "Signup to create Pass!")
+    }
+  } catch (err) {
+    next(err)
+  }
+})
+
+//@desc View all passes generated by a user
+//@route Get /users/viewPasses
+const viewPasses = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
```

Comment:
Because we need to access the database in order to perform each and every operation, if it is generating a pass then we need User's information, if it's verifying a pass then too we need to access the database and so on. So we need the database connection for each operation (also these three operations are independent so every time we need the database connection)

Reply:
Isn't this should be same for login and signup but the statement is not in those function?

And can you check the current function without having this `connectDB()` line?

Reply:
For signup it is not needed as we are not needing any data from the database for executing that request, we are signing a new user. For login, i think it should have been there but idk how's it working without it...
Without connectDB(), internal server error is displayed

Reply:
Lets connect for this tomorrow.

Reply:
okay